src/simd.rs

```diff
@@ -721,17 +698,17 @@ mod test {
 
         // Allocate a 16-byte string at page boundary.  To verify this
         // test, set protect=false to prevent segfaults.
-        let text = alloc_guarded_string("0123456789abcdef", true);
```

Comment:
Hmm... it didn't seem to be ever used?

Reply:
Ping.

Reply:
Right. It's a manual flip at the moment. Causing the test to fail usually results in a segfault, which can't be caught by the test harness, so it's not possible to write a test that asserts the bad behavior.

Reply:
Right, but it used to always `protect`, so are you saying you used to manually disable protection? I don't see how it would make a test fail, rather the opposite.

I'm happy to reintroduce the flag, just trying to figure out in which case you *wouldn't* want the protection to be applied.

Reply:
> so are you saying you used to manually disable protection

Yes, that's what I meant.

Honestly, I don't fully recall when I wanted to turn it off (you can see it was there from the very first PR #1), so I'm ok with waiting until I remember and then re-adding it.